tests/blob-server.js

```diff
@@ -215,6 +217,52 @@ test('GET photo returns 404 when trying to get non-replicated blob', async (t) =
   t.is(res.statusCode, 404)
 })
 
+test('GET photo returns 404 when trying to get non-existent blob', async (t) => {
```

Comment:
maybe out of scope but would be helpful to ensure that the reasons for the 404s align with our expectations. right now this test only checks the status code but maybe in the short term, check the status message as well?

more futureproof solution would be to define some custom fastify errors to use

Reply:
Just to point out that since this server is serving images, error messages do not make a lot of sense, because we are not processing the response in JS code (the response is handled on the platform specific image component). For now I think 404 is ok.